esp/services/ws_sql/ws_sqlService.cpp

```diff
@@ -1089,10 +1089,13 @@ void CwssqlEx::createWUXMLParams(StringBuffer & xmlparams, const IArrayOf <ISQLE
         if (exp->getExpType() == Value_ExpressionType)
         {
             SQLValueExpression * currentvalplaceholder = dynamic_cast<SQLValueExpression *>(exp);
-            currentvalplaceholder->trimTextQuotes();
```

Comment:
In this and other cases, it seems to me that you are fixing the symptom (that Coverity complains your code is not safe, as a dynamic cast may return null) rather than the real bug. Yes, your change will suppress the warning, and yes, if the dynamic cast did return null then you will now skip some code rather than coring. But you've already checked the expression type, so if the dynamic cast DOES return null then it implies something has gone really wrong somewhere, and simply ignoring that fact and continuing silently is not really the correct behaviour.

Seems to me that the more correct fix (which would also improve efficiency - dynamic cast is very slow) is to change the dynamic cast to a static one.

Reply:
In agreement, will revise